go-controller/pkg/ovn/egressgw.go

```diff
@@ -408,40 +409,28 @@ func (oc *Controller) deletePodGWRoutesForNamespace(pod *kapi.Pod, namespace str
 		return
 	}
 
+	gws := sets.NewString()
 	for _, gwIP := range foundGws.gws {
-		// check for previously configured pod routes
-		routeInfos := oc.getRouteInfosForNamespace(namespace)
-		for _, routeInfo := range routeInfos {
-			routeInfo.Lock()
-			if routeInfo.deleted {
-				routeInfo.Unlock()
-				continue
-			}
-			for podIP, route := range routeInfo.podExternalRoutes {
-				for routeGwIP, gr := range route {
-					if gwIP.String() == routeGwIP {
-						oc.deleteOnePodGWRoute(routeInfo, podIP, gwIP.String(), gr)
-					}
-				}
-			}
-			routeInfo.Unlock()
-		}
+		gws.Insert(gwIP.String())
 	}
+	oc.deleteGWRoutesForNamespace(namespace, gws)
 }
 
-// deleteGwRoutesForNamespace handles deleting all routes to gateways for a pod on a specific GR
-func (oc *Controller) deleteGWRoutesForNamespace(namespace string) {
-	// TODO(trozet): batch all of these with ebay bindings
-	routeInfos := oc.getRouteInfosForNamespace(namespace)
-	for _, routeInfo := range routeInfos {
+// deleteGwRoutesForNamespace handles deleting routes to gateways for a pod on a specific GR.
+// If matching gateways to delete routes for are given, all routes for the namespace are
```

Comment:
if matching gateways to delete routes for are 0, then all routes for the namespace are deleted?

Reply:
Clarified per your text.